pkg/oc/util/tokencmd/negotiator_sspi.go

```diff
@@ -0,0 +1,188 @@
+// +build windows
+
+package tokencmd
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	"github.com/alexbrainman/sspi"
+	"github.com/alexbrainman/sspi/negotiate"
+	"github.com/golang/glog"
+)
+
+const (
+	// sane set of default flags, see sspiNegotiator.flags
+	// TODO make configurable?
+	flags = sspi.ISC_REQ_CONFIDENTIALITY |
+		sspi.ISC_REQ_INTEGRITY |
+		sspi.ISC_REQ_MUTUAL_AUTH |
+		sspi.ISC_REQ_REPLAY_DETECT |
+		sspi.ISC_REQ_SEQUENCE_DETECT
+
+	// separator used in fully qualified user name format
+	domainSeparator = `\`
+
+	// max lengths for various fields, see sspiNegotiator.principalName
+	maxUsername = 256
+	maxPassword = 256
+	maxDomain   = 15
+)
+
+func SSPIEnabled() bool {
+	return true
+}
+
+// sspiNegotiator handles negotiate flows on windows via SSPI
+// It expects sspiNegotiator.InitSecContext to be called until sspiNegotiator.IsComplete returns true
+type sspiNegotiator struct {
+	// optional DOMAIN\Username and password
+	// https://msdn.microsoft.com/en-us/library/windows/desktop/aa374714(v=vs.85).aspx
+	// https://msdn.microsoft.com/en-us/library/windows/desktop/aa380131(v=vs.85).aspx
+	// pAuthData [in]: If credentials are supplied, they are passed via a pointer to a sspi.SEC_WINNT_AUTH_IDENTITY
+	// structure that includes those credentials.
+	// When using the Negotiate package, the maximum character lengths for user name, password, and domain are
+	// 256, 256, and 15, respectively.
+	principalName string
```

Comment:
Instead of adding this long comment here I would define a self contained function that deal with validating a principal name, with the comment and normative references as comments in that function and no module level constants.

Reply:
I still think you should address this one